substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/elf/dwarf/DwarfLineSectionImpl.java

```diff
@@ -320,7 +320,7 @@ private int writeHeader(ClassEntry classEntry, byte[] buffer, int p) {
         /*
          * 4 ubyte prologue length includes rest of header and dir + file table section.
          */
-        int prologueSize = classEntry.getLinePrologueSize() - 6;
+        int prologueSize = classEntry.getLinePrologueSize() - 10;
```

Comment:
So the 10 here is from 4 (length field)  + 2 (version field) + 4 (prologue length field), right?

Reply:
Yes, that's right and originally I failed to exclude the 4 byte slot for for the prologue field from the prologue length.

I thought about using a private static final field for this value here to allow the above to be documented in some javadoc. However, I thought that would be overkill as it was fairly obvious at this point that we had just written 10 bytes.

I also thought about asserting at the end of writing the header that the sum of the header size, file table size and directory size added up to the prologue field value + 10. However, that would involve reading back an int out of the byte array at offset 6 in order to validate what was fairly clearly written using that same computation. Once again that seemed to be overkill (so far there has been no need for any methods to read structured data from the byte[] so adding one to implement an assert seemed to be a step too far).

If you want I couled revise either or both those decisions?

Reply:
Nah ... just use `- (4 + 2 + 4)` instead of `-10`.

Reply:
Done.